hashes/src/sha256.rs

```diff
@@ -929,22 +949,25 @@ mod tests {
         ]);
         // 32 bytes of zeroes representing "new asset"
         engine.input(&[0; 32]);
+
+        let total_bytes_hashed = 64;
+
         assert_eq!(
-            engine.midstate(),
+            engine.midstate().expect("total_bytes_hashed is valid"),
             // RPC output
-            sha256::Midstate::from_byte_array([
+            sha256::Midstate::new([
                 0x0b, 0xcf, 0xe0, 0xe5, 0x4e, 0x6c, 0xc7, 0xd3,
                 0x4f, 0x4f, 0x7c, 0x1d, 0xf0, 0xb0, 0xf5, 0x03,
                 0xf2, 0xf7, 0x12, 0x91, 0x2a, 0x06, 0x05, 0xb4,
                 0x14, 0xed, 0x33, 0x7f, 0x7f, 0x03, 0x2e, 0x03,
-            ])
+            ], total_bytes_hashed)
         );
     }
 
     #[test]
     fn engine_with_state() {
         let mut engine = sha256::Hash::engine();
-        let midstate_engine = sha256::HashEngine::from_midstate(engine.midstate(), 0);
+        let midstate_engine = sha256::HashEngine::from_midstate(engine.midstate_unchecked());
```

Comment:
Why is this not `.midstate().unwrap()`?

Reply:
I read this as randomly/adhoc choosing which one to call in the tests, so that both are covered.

Reply:
The checked version calls into unchecked so it shouldn't be needed? Also it's better to test public API than private (but the private will probably stay)